z3-sys/build.rs

```diff
@@ -1,49 +1,78 @@
 
-fn main() {
-    #[cfg(feature = "static-link-z3")]
-    build_z3();
+#[cfg(not(feature = "dynamic-link-z3"))]
+#[cfg(not(feature = "static-link-z3"))]
+const Z3_RELEASE: &str = "z3-4.8.12";
+
+#[cfg(target_os = "windows")]
+const Z3_ARCH: &str = "x64-win";
+#[cfg(target_os = "linux")]
+const Z3_ARCH: &str = "x64-glibc-2.31";
 
+fn main() {
     println!("cargo:rerun-if-changed=build.rs");
+    get_z3();
+    #[cfg(feature = "use-bindgen")]
+    bindgen();
+}
 
-    let header = if cfg!(feature = "static-link-z3") {
-        "z3/src/api/z3.h".to_string()
-    } else if let Ok(header_path) = std::env::var(Z3_HEADER_VAR) {
-        header_path
-    } else {
-        "wrapper.h".to_string()
-    };
-    println!("cargo:rerun-if-env-changed={}", Z3_HEADER_VAR);
-    println!("cargo:rerun-if-changed={}", header);
-    let out_path = std::path::PathBuf::from(std::env::var("OUT_DIR").unwrap());
+#[cfg(not(feature = "dynamic-link-z3"))]
+#[cfg(not(feature = "static-link-z3"))]
+/// Use precompiled Z3 binaries.
+fn get_z3() {
+    let out_dir = std::path::PathBuf::from(std::env::var("OUT_DIR").unwrap());
+    let archive_filename = format!("{}-{}", Z3_RELEASE, Z3_ARCH);
+    let archive_url = format!("https://github.com/Z3Prover/z3/releases/download/{}/{}.zip", Z3_RELEASE, archive_filename);
+    let archive_path = out_dir.join(format!("{}.zip", archive_filename));
+    let release_folder = out_dir.join(archive_filename);
 
-    for x in &[
-        "ast_kind",
-        "ast_print_mode",
-        "decl_kind",
-        "error_code",
-        "goal_prec",
-        "param_kind",
-        "parameter_kind",
-        "sort_kind",
-        "symbol_kind",
-    ] {
-        let enum_bindings = bindgen::Builder::default()
-            .header(&header)
-            .parse_callbacks(Box::new(bindgen::CargoCallbacks))
-            .generate_comments(false)
-            .rustified_enum(format!("Z3_{}", x))
-            .allowlist_type(format!("Z3_{}", x))
-            .generate()
-            .expect("Unable to generate bindings");
-        enum_bindings
-            .write_to_file(out_path.join(format!("{}.rs", x)))
-            .expect("Couldn't write bindings!");
-    }
+    let curl = std::process::Command::new("curl")
+        .arg("-L")
+        .arg("-o")
+        .arg(&archive_path.to_str().unwrap())
+        .arg(&archive_url)
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(curl.success());
+
+    #[cfg(target_os = "windows")]
+    let tar = std::process::Command::new("tar")
+        .current_dir(archive_path.parent().unwrap())
+        .arg("-x")
+        .arg("-f")
+        .arg(&archive_path.to_str().unwrap())
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+
+    #[cfg(not(target_os = "windows"))]
+    let tar = std::process::Command::new("unzip")
+        .current_dir(archive_path.parent().unwrap())
+        .arg(&archive_path.to_str().unwrap())
+        .spawn()
+        .unwrap()
```

Comment:
The error message here can be a bit confusing if one doesn't have the `unzip` binary installed, as the error message shown is just that a file somewhere is missing.

Reply:
would it be ok to introduce a new dev dep that handles that stuff?